diff --git a/autostart/CMakeLists.txt b/autostart/CMakeLists.txt
index 76cacf8..3863cbe 100644
--- a/autostart/CMakeLists.txt
+++ b/autostart/CMakeLists.txt
@@ -10,4 +10,4 @@ lxqt_translate_desktop(DESKTOP_FILES
 add_custom_target(autostart_desktop_files ALL DEPENDS ${DESKTOP_FILES})
 #************************************************
 
-install(FILES ${DESKTOP_FILES} DESTINATION /etc/xdg/autostart)
+install(FILES ${DESKTOP_FILES} DESTINATION "${LXQT_ETC_XDG_DIR}/autostart")
diff --git a/xsession/CMakeLists.txt b/xsession/CMakeLists.txt
index 9bd6461..37b8837 100644
--- a/xsession/CMakeLists.txt
+++ b/xsession/CMakeLists.txt
@@ -18,8 +18,8 @@ ENDMACRO(INSTALL_SESSION_FILES)
 
 #INSTALL_SESSION_FILES("/etc/X11/sessions")
 #INSTALL_SESSION_FILES("/usr/share/sessions")
-INSTALL_SESSION_FILES("/usr/share/xsessions")
-INSTALL_SESSION_FILES("/usr/share/apps/kdm/sessions")
+INSTALL_SESSION_FILES("${CMAKE_INSTALL_PREFIX}/${CMAKE_INSTALL_DATAROOTDIR}/xsessions")
+INSTALL_SESSION_FILES("${CMAKE_INSTALL_PREFIX}/${CMAKE_INSTALL_DATAROOTDIR}/apps/kdm/sessions")
 #INSTALL_SESSION_FILES("/usr/local/share/apps/kdm/sessions")
 #INSTALL_SESSION_FILES("/usr/share/config/kdm/sessions")
 #INSTALL_SESSION_FILES("/usr/local/share/config/kdm/sessions")
