diff -ru3 glibc-2.23-orig/sysdeps/generic/dl-cache.h glibc-2.23/sysdeps/generic/dl-cache.h
--- glibc-2.23-orig/sysdeps/generic/dl-cache.h	2016-06-05 15:37:27.931125442 +0300
+++ glibc-2.23/sysdeps/generic/dl-cache.h	2016-06-05 15:44:38.210116662 +0300
@@ -28,7 +28,7 @@
 #endif
 
 #ifndef LD_SO_CACHE
-# define LD_SO_CACHE SYSCONFDIR "/ld.so.cache"
+# define LD_SO_CACHE SYSCONFDIR "/ld-nix.so.cache"
 #endif
 
 #ifndef add_system_dir
