diff --git a/utils.go b/utils.go
index 1c32b93..5487d26 100644
--- a/utils.go
+++ b/utils.go
@@ -602,25 +602,29 @@ func getLibModMounts() ([]specs.Mount, error) {
 		return nil, err
 	}
 
-	path := filepath.Join("/lib/modules/", kernelRel)
-	if _, err := os.Stat(path); err != nil {
-		return nil, err
+	srcPath := filepath.Join("/lib/modules/", kernelRel)
+	if _, err := os.Stat(srcPath); err != nil {
+		srcPath = filepath.Join("/run/booted-system/kernel-modules/lib/modules/", kernelRel)
+		if _, err := os.Stat(srcPath); err != nil {
+			return nil, err
+		}
 	}
 
 	mounts := []specs.Mount{}
 
 	// Do *not* follow symlinks as they normally point to the linux headers which we
 	// mount also (see getLinuxHeadersMount()).
+	destPath := filepath.Join("/lib/modules/", kernelRel)
 	mounts, err = createMountSpec(
-		path,
-		path,
+		srcPath,
+		destPath,
 		"bind",
 		[]string{"ro", "rbind", "rprivate"}, false, []string{"/usr/src"},
 	)
 
 	if err != nil {
 		return nil, fmt.Errorf("failed to create mount spec for linux modules at %s: %v",
-			path, err)
+			destPath, err)
 	}
 
 	return mounts, nil
